controllers/k6_controller.go

```diff
@@ -93,5 +92,8 @@ func (r *K6Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.K6{}).
 		Owns(&batchv1.Job{}).
+		Watches(&source.Kind{Type: &v1.Pod{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: &K6PodsWatchMap{log: mgr.GetLogger()}},
+			builder.WithPredicates(filterK6Pods())).
```

Comment:
Based on my understanding.

- For(**<ownerType>**) - Tells the manager to trigger the Reconcile(...) whenever any object of **<ownerType>** is created/updated/deleted.
 
- Owns(**<childType>**) - Tells the manager to trigger the Reconcile(...) whenever any object of **<childType>** having an ownerReference to **<ownerType>** is created/updated/deleted.
 
- Watches() - We can configure any possible event happening in the cluster to trigger the controller’s Reconcile() method.

This is the key point to remove the current wait.Pull() and force the Reconcile() in case we receive any k6 Pod event.

As we discussed, we are using the operator with this changes and all resources are being clearing as expected :)

Reply:
Update on this point. Absent clean up I observed last week was likely due to an issue on my side: clean up works in another setup :+1: So resolving this point.